File: tempconv/tempconv.php
<?php

const AbsoluteZeroC = -273.15;
const FreezingC = 0;
const BoilingC = 100;

function Celsius($c) {
    return $c;
}

function Fahrenheit($f) {
    return $f;
}

function CToF($c) {
    return Fahrenheit($c * 9 / 5 + 32);
}

function FToC($f) {
    return Celsius(($f - 32) * 5 / 9);
}

?>

File: tempconv/tempconv.c
#include <stdio.h>

typedef double Celsius;
typedef double Fahrenheit;

const Celsius AbsoluteZeroC = -273.15;
const Celsius FreezingC = 0;
const Celsius BoilingC = 100;

void printCelsius(Celsius c)
{
    printf("%.2f°C\n", c);
}

void printFahrenheit(Fahrenheit f)
{
    printf("%.2f°F\n", f);
}

Fahrenheit celsiusToFahrenheit(Celsius c)
{
    return c * 9 / 5 + 32;
}

Celsius fahrenheitToCelsius(Fahrenheit f)
{
    return (f - 32) * 5 / 9;
}
